test/test_credentials.py

```diff
@@ -0,0 +1,100 @@
+# coding=utf-8
+import tempfile
+import unittest
+from pathlib import Path
+
+from credentials import CredentialsStore
+
+ACCESS_TOKEN = "00000000-0000-0000-0000-000000000000"
+
+
+class TestCredentialsStore(unittest.TestCase):
+
+    def setUp(self):
+        self.credentials_dir = Path(tempfile.NamedTemporaryFile().name)
```

Comment:
You can use a context manager for this:

```python
# create a temporary file using a context manager
with tempfile.TemporaryFile() as fp:
    fp.write(b'Hello world!')
    fp.seek(0)
    fp.read()
```
https://docs.python.org/3.7/library/tempfile.html#tempfile-examples

To apply this to your tests you can use a decorator:

```python
def with_credential_store(function):
    def wrapper(self):
        with tempfile.NamedTemporaryFile() as path:
            token_path = Path(path) / 'tokens.txt'
            store = CredentialsStore(credentials_path=token_path)
            function(self, store)
    return wrapper
```

You would then apply this like so:

```python
@with_credential_store
def test_file_created_on_read(self, store):
    ...
```

Reply:
Alternatively you don't have to care about destroying it as the contract states:

> It will be destroyed as soon as it is closed (including an implicit close when the object is garbage collected).

Reply:
Neat. Thanks